pkg/kafka/consumer/config.go

```diff
@@ -15,26 +15,39 @@
 package consumer
 
 import (
-	"io"
-
+	"context"
 	"github.com/Shopify/sarama"
-	"github.com/bsm/sarama-cluster"
-
 	"github.com/jaegertracing/jaeger/pkg/kafka/auth"
 )
 
-// Consumer is an interface to features of Sarama that are necessary for the consumer
-type Consumer interface {
-	Partitions() <-chan cluster.PartitionConsumer
-	MarkPartitionOffset(topic string, partition int32, offset int64, metadata string)
-	io.Closer
-}
-
 // Builder builds a new kafka consumer
 type Builder interface {
 	NewConsumer() (Consumer, error)
 }
 
+type Consumer interface {
+	Consume(ctx context.Context, handler sarama.ConsumerGroupHandler) error
```

Comment:
consume what? please document new interfaces and methods

Reply:
This starts the consumer loop , which consume the kafka partitions, I'll document it, but escentialy this is an interface that wraps ` sarama.ConsumerGroup`, so I borrow the names from the ConsumerGroup interface.